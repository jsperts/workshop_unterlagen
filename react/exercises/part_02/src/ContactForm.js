import React from 'react';
import './ContactForm.css';

function ContactForm() {
  return (
    <form className="contact-form">
      <div className="row">
        <div className="col-xs-6">
          <div className="form-group">
            <label htmlFor="name">Name:</label>
            <input type="text" className="form-control" name="name" id="name" />
          </div>
          <div className="form-group">
            <label htmlFor="telephone">Telephone:</label>
            <input
              type="text"
              className="form-control"
              name="telephone"
              id="telephone"
            />
          </div>
        </div>
        <div className="col-xs-6">
          <div className="form-group">
            <label htmlFor="email">Email:</label>
            <input
              type="text"
              className="form-control"
              name="email"
              id="email"
            />
          </div>
          <div className="form-group">
            <label htmlFor="type">Type:</label>
            <select className="form-control" name="type" id="type">
              {['Private', 'Business'].map(opt => (
                <option key={opt} value={opt}>
                  {opt}
                </option>
              ))}
            </select>
          </div>
        </div>
      </div>
      <div className="text-center">
        <div className="btn-group">
          <button type="submit" className="btn btn-primary">
            Add
          </button>
          <button type="button" className="btn btn-secondary">
            Cancel
          </button>
        </div>
      </div>
    </form>
  );
}

export default ContactForm;
