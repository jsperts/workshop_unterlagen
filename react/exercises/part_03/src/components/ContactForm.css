.contact-form {
  background-color: #f8f8f8;
  border: 1px solid #e7e7e7;
  border-radius: 4px;
  margin-bottom: 20px;
  padding: 10px;
}
