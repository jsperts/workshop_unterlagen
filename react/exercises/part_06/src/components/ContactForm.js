import React from 'react';
import PropTypes from 'prop-types';

import './ContactForm.css';

function ContactForm({ onCancel, onSubmit, onUpdateField, formData }) {
  return (
    <form onSubmit={onSubmit} className="contact-form">
      <div className="row">
        <div className="col-xs-6">
          <div className="form-group">
            <label htmlFor="name">Name:</label>
            <input
              type="text"
              className="form-control"
              name="name"
              id="name"
              value={formData.name}
              onChange={e => onUpdateField('name', e.target.value)}
            />
          </div>
          <div className="form-group">
            <label htmlFor="tel">Telephone:</label>
            <input
              type="text"
              className="form-control"
              name="tel"
              id="tel"
              value={formData.tel}
              onChange={e => onUpdateField('tel', e.target.value)}
            />
          </div>
        </div>
        <div className="col-xs-6">
          <div className="form-group">
            <label htmlFor="email">Email:</label>
            <input
              type="text"
              className="form-control"
              name="email"
              id="email"
              value={formData.email}
              onChange={e => onUpdateField('email', e.target.value)}
            />
          </div>
          <div className="form-group">
            <label htmlFor="type">Type:</label>
            <select
              className="form-control"
              name="type"
              id="type"
              value={formData.type}
              onChange={e => onUpdateField('type', e.target.value)}
            >
              {['Private', 'Business'].map(opt => (
                <option key={opt} value={opt}>
                  {opt}
                </option>
              ))}
            </select>
          </div>
        </div>
      </div>
      <div className="text-center">
        <div className="btn-group">
          <button type="submit" className="btn btn-primary">
            Add
          </button>
          <button
            type="button"
            className="btn btn-secondary"
            onClick={onCancel}
          >
            Cancel
          </button>
        </div>
      </div>
    </form>
  );
}

ContactForm.propTypes = {
  onCancel: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onUpdateField: PropTypes.func.isRequired,
  formData: PropTypes.shape({
    name: PropTypes.string,
    email: PropTypes.string,
    tel: PropTypes.string,
    type: PropTypes.oneOf(['Private', 'Business']),
    id: PropTypes.number,
  }).isRequired,
};

export default ContactForm;
