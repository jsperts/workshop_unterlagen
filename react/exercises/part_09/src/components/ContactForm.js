import React from 'react';
import PropTypes from 'prop-types';

import './ContactForm.css';
import { FormData as FormDataPropType } from './propTypes';

function ContactForm({
  onCancel,
  onSubmit,
  onUpdateField,
  formData,
  submitButtonLabel,
}) {
  return (
    <form onSubmit={onSubmit} className="contact-form">
      <div className="row">
        <div className="col-xs-6">
          <div className="form-group">
            <label htmlFor="name">Name:</label>
            <input
              type="text"
              className="form-control"
              name="name"
              id="name"
              value={formData.name}
              onChange={e => onUpdateField('name', e.target.value)}
            />
          </div>
          <div className="form-group">
            <label htmlFor="tel">Telephone:</label>
            <input
              type="text"
              className="form-control"
              name="tel"
              id="tel"
              value={formData.tel}
              onChange={e => onUpdateField('tel', e.target.value)}
            />
          </div>
        </div>
        <div className="col-xs-6">
          <div className="form-group">
            <label htmlFor="email">Email:</label>
            <input
              type="text"
              className="form-control"
              name="email"
              id="email"
              value={formData.email}
              onChange={e => onUpdateField('email', e.target.value)}
            />
          </div>
          <div className="form-group">
            <label htmlFor="type">Type:</label>
            <select
              className="form-control"
              name="type"
              id="type"
              value={formData.type}
              onChange={e => onUpdateField('type', e.target.value)}
            >
              {['Private', 'Business'].map(opt => (
                <option key={opt} value={opt}>
                  {opt}
                </option>
              ))}
            </select>
          </div>
        </div>
      </div>
      <div className="text-center">
        <div className="btn-group">
          <button type="submit" className="btn btn-primary">
            {submitButtonLabel}
          </button>
          <button
            type="button"
            className="btn btn-secondary"
            onClick={onCancel}
          >
            Cancel
          </button>
        </div>
      </div>
    </form>
  );
}

ContactForm.propTypes = {
  onCancel: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onUpdateField: PropTypes.func.isRequired,
  submitButtonLabel: PropTypes.string.isRequired,
  formData: PropTypes.shape(FormDataPropType).isRequired,
};

export default ContactForm;
