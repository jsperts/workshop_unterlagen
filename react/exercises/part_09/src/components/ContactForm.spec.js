import React from 'react';
import { shallow } from 'enzyme';

import ContactForm from './ContactForm';

describe('ContactForm', () => {
  test('should match the snapshot', () => {
    const props = {
      onCancel() {},
      onSubmit() {},
      onUpdateField() {},
      submitButtonLabel: 'foo',
      formData: {
        name: 'a',
        email: 'b',
        tel: 'c',
        type: 'Private',
        id: 1
      },
    };
    const wrapper = shallow(<ContactForm {...props} />);
    expect(wrapper.getElements()).toMatchSnapshot();
  });
});
