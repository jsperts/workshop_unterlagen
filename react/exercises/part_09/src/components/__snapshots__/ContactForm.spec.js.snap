// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm should match the snapshot 1`] = `
Array [
  <form
    className="contact-form"
    onSubmit={[Function]}
>
    <div
        className="row"
    >
        <div
            className="col-xs-6"
        >
            <div
                className="form-group"
            >
                <label
                    htmlFor="name"
                >
                    Name:
                </label>
                <input
                    className="form-control"
                    id="name"
                    name="name"
                    onChange={[Function]}
                    type="text"
                    value="a"
                />
            </div>
            <div
                className="form-group"
            >
                <label
                    htmlFor="tel"
                >
                    Telephone:
                </label>
                <input
                    className="form-control"
                    id="tel"
                    name="tel"
                    onChange={[Function]}
                    type="text"
                    value="c"
                />
            </div>
        </div>
        <div
            className="col-xs-6"
        >
            <div
                className="form-group"
            >
                <label
                    htmlFor="email"
                >
                    Email:
                </label>
                <input
                    className="form-control"
                    id="email"
                    name="email"
                    onChange={[Function]}
                    type="text"
                    value="b"
                />
            </div>
            <div
                className="form-group"
            >
                <label
                    htmlFor="type"
                >
                    Type:
                </label>
                <select
                    className="form-control"
                    id="type"
                    name="type"
                    onChange={[Function]}
                    value="Private"
                >
                    <option
                        value="Private"
                    >
                        Private
                    </option>
                    <option
                        value="Business"
                    >
                        Business
                    </option>
                </select>
            </div>
        </div>
    </div>
    <div
        className="text-center"
    >
        <div
            className="btn-group"
        >
            <button
                className="btn btn-primary"
                type="submit"
            >
                foo
            </button>
            <button
                className="btn btn-secondary"
                onClick={[Function]}
                type="button"
            >
                Cancel
            </button>
        </div>
    </div>
</form>,
]
`;
